src/mono/wasm/build/WasmApp.targets

```diff
@@ -134,7 +134,33 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="_WasmGenerateAppBundle" Condition="'$(WasmGenerateAppBundle)' == 'true'">
+  <Target Name="_WasmGenerateRuntimeConfig" Condition="'$(OutputType)' != 'library'">
+    <PropertyGroup>
+      <_MainAssemblyPath Condition="'%(WasmAssembliesToBundle.FileName)' == $(AssemblyName) and '%(WasmAssembliesToBundle.Extension)' == '.dll'">%(WasmAssembliesToBundle.Identity)</_MainAssemblyPath>
+      <RuntimeConfigFilePath>$([System.IO.Path]::ChangeExtension($(_MainAssemblyPath), '.runtimeconfig.json'))</RuntimeConfigFilePath>
+      <ParsedRuntimeConfigFilePath>$([System.IO.Path]::GetDirectoryName($(_MainAssemblyPath)))\runtimeconfig.bin</ParsedRuntimeConfigFilePath>
+    </PropertyGroup>
+
+    <Error Condition="!Exists('$(RuntimeConfigFilePath)')" Text="RuntimeConfigFilePath=$(RuntimeConfigFilePath) doesn't exist" />
+
+    <ItemGroup>
+      <RuntimeConfigReservedProperties Include="RUNTIME_IDENTIFIER"/>
+      <RuntimeConfigReservedProperties Include="APP_CONTEXT_BASE_DIRECTORY"/>
+    </ItemGroup>
+
+    <!-- Parse *.runtimeconfig.json file -->
+    <RuntimeConfigParserTask
+        RuntimeConfigFile="$(RuntimeConfigFilePath)"
+        OutputFile="$(ParsedRuntimeConfigFilePath)"
+        RuntimeConfigReservedProperties="@(RuntimeConfigReservedProperties)">
+    </RuntimeConfigParserTask>
+
+    <ItemGroup>
+      <WasmFilesToIncludeInFileSystem Include="$(ParsedRuntimeConfigFilePath)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_WasmGenerateAppBundle" Condition="'$(WasmGenerateAppBundle)' == 'true'" DependsOnTargets="_WasmGenerateRuntimeConfig">
```

Comment:
If this is intended to work in Blazor this doesn't look sufficient since _WasmGenerateAppBundle does not run in that case.

Reply:
Where is `WasmAppBuilder` called for Blazor?

Reply:
Blazor doesn't use `WasmAppBuilder`. It has its own packaging/bundling stuff.

Reply:
Where does that logic live?

Reply:
Created an issue to track this. https://github.com/dotnet/aspnetcore/issues/35010

Reply:
There is a broader problem with making this work in blazor in that it requires a workload and the baseline blazor scenarios are supposed to be functional without any workloads installed.

Reply:
Created a tracking issue https://github.com/dotnet/aspnetcore/issues/35010